packages/jsx/src/utils.test.ts

```diff
@@ -0,0 +1,75 @@
+import { atom, Atom, AtomMaybe, CtxSpy, isAtom } from '@reatom/core'
+import { createTestCtx } from '@reatom/testing'
+import * as assert from 'uvu/assert'
+import { cn } from './utils'
+
+describe('reatomClassName', () => {
+  const ctx = createTestCtx()
+
+  it('handles falsy correctly', () => {
+    assert.is(ctx.get(cn(false)), '')
+    assert.is(ctx.get(cn(true)), '')
+    assert.is(ctx.get(cn(null)), '')
+    assert.is(ctx.get(cn(undefined)), '')
+    assert.is(ctx.get(cn(atom(undefined))), '')
+    assert.is(ctx.get(cn(() => undefined)), '')
+  })
+
+  it('handles falsy object correctly', () => {
+    assert.is(ctx.get(cn({
+      a: '',
+      b: 0,
+      c: NaN,
+      d: false,
+      e: null,
+      f: undefined,
+      g: atom(undefined),
+    })), '')
+  })
+
+  it('handles falsy array correctly', () => {
+    assert.is(ctx.get(cn([
+      '',
+      null,
+      undefined,
+      {},
+      [],
+      atom(undefined),
+      () => undefined,
+    ])), '')
+  })
+
+  it('handles object correctly', () => {
+    assert.is(ctx.get(cn({
+      a: 'a',
+      b: 1,
+      c: true,
+      d: {},
+      e: [],
+      f: atom(true),
+      g: () => undefined,
+    })), 'a b c d e f g ')
+  })
+
+  it('handles deep array correctly', () => {
+    assert.is(ctx.get(cn(['a', ['b', ['c']]])), 'a b c   ')
+  })
+
+  it('handles deep atom correctly', () => {
+    assert.is(ctx.get(cn(atom(() => atom(() => atom('a'))))), 'a')
+  })
+
+  it('handles deep getter correctly', () => {
+    assert.is(ctx.get(cn(() => () => () => 'a')), 'a')
+  })
+
+  it('handles complex correctly', () => {
+    assert.is(ctx.get(cn(() => atom(() => [
+      'a',
+      {b: atom(true)},
+      ['c'],
+      atom('d'),
+      () => 'e',
+    ]))), 'a b  c  d e ')
+  })
+})
```

Comment:
Please, test the change behaviour of created atom to ensure that the created atom correctly subscribes (spy) to nested atom. You can change one of the existing tests